rotkehlchen/chain/ethereum/modules/sushiswap/decoder.py

```diff
@@ -112,6 +124,11 @@ def decoding_rules(self) -> list[Callable]:
             self._maybe_decode_v2_liquidity_addition_and_removal,
         ]
 
+    def post_decoding_rules(self) -> list[tuple[int, Callable]]:
+        return [
+            (0, self._fix_order_of_events_for_swaps),
+        ]
```

Comment:
There is no need to add this post decoding rule for all three. You can check in the function that if the counterparty is any of the three we said make sure that the swaps are consecutive. This way a transactions will loop thrice trough this logic for no reason